src/main/scala/no/ndla/frontpageapi/service/WriteService.scala

```diff
@@ -19,26 +19,51 @@ trait WriteService {
 
   class WriteService {
 
-    def newSubjectPage(subject: api.NewOrUpdateSubjectFrontPageData): Try[api.SubjectPageData] = {
+    def newSubjectPage(subject: api.NewSubjectFrontPageData): Try[api.SubjectPageData] = {
       for {
         convertedSubject <- ConverterService.toDomainSubjectPage(subject)
         subjectPage <- subjectPageRepository.newSubjectPage(convertedSubject, subject.externalId)
-      } yield ConverterService.toApiSubjectPage(subjectPage, "nb")
+        converted <- ConverterService.toApiSubjectPage(subjectPage, "nb")
+      } yield converted
     }
 
-    def updateSubjectPage(id: Long, subject: api.NewOrUpdateSubjectFrontPageData): Try[api.SubjectPageData] = {
+    def updateSubjectPage(id: Long, subject: api.NewSubjectFrontPageData): Try[api.SubjectPageData] = {
       subjectPageRepository.exists(id) match {
         case Success(exists) if exists =>
           for {
             domainSubject <- ConverterService.toDomainSubjectPage(id, subject)
             subjectPage <- subjectPageRepository.updateSubjectPage(domainSubject)
-          } yield ConverterService.toApiSubjectPage(subjectPage, "nb")
+            converted <- ConverterService.toApiSubjectPage(subjectPage, "nb")
+          } yield converted
         case Success(_) =>
           Failure(NotFoundException(id))
         case Failure(ex) => Failure(ex)
       }
     }
 
+    def updateSubjectPage(id: Long, subject: api.UpdatedSubjectFrontPageData): Try[api.SubjectPageData] = {
+      subjectPageRepository.withId(id) match {
+        case Some(existingSubject) =>
+          for {
+            domainSubject <- ConverterService.toDomainSubjectPage(existingSubject, subject)
+            subjectPage <- subjectPageRepository.updateSubjectPage(domainSubject)
+            converted <- ConverterService.toApiSubjectPage(subjectPage, subject.about.get.head.language)
+          } yield converted
+        case None if subjectPageRepository.exists(id).getOrElse(false) =>
+          newFromUpdatedSubjectPage(subject) match {
+            case Failure(ex)             => Failure(ex)
+            case Success(newSubjectPage) => updateSubjectPage(id, newSubjectPage)
+          }
+        case None =>
+          Failure(NotFoundException(404))
```

Comment:
Litt missvisende og alltid returnere at id `404` ikke finnes i feilmeldingen :smile: 
```suggestion
          Failure(NotFoundException(id))
```